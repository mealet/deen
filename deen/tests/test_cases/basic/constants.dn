fn main() i32 {
  // integer constants
  let decimal = 1000;
  let decimal_sep = 1_000;
  let binary = 0b1111101000;
  let hexadecimal = 0x3E8;
  let hexadecimal_lc = 0x3e8;

  println!("{}", decimal);
  println!("{}", decimal_sep);
  println!("{}", binary);
  println!("{}", hexadecimal);
  println!("{}", hexadecimal_lc);

  // float constants
  let default = 1.0;
  let bigger = 51.32;
  let pi = 3.141592;

  println!("{}", default);
  println!("{}", bigger);
  println!("{}", pi);

  // boolean constants

  let boolean_true = true;
  let boolean_false = false;

  println!("{} -> {} | {} -> {}", boolean_true, cast!(boolean_true, u32), boolean_false, cast!(boolean_false, u32));

  // character constants
  
  let character: char = 'a';
  println!("{}", character);

  // string constants

  let raw_string: *char = "hello";
  println!("{}", raw_string);

  return 0;
}
