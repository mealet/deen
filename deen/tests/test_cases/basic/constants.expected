1000
1000
1000
1000
1000
1.000000
51.320000
3.141592
true -> 1 | false -> 0
a
hello
